apid.go

```diff
@@ -69,6 +76,24 @@ func InitializePlugins() {
 	log.Debugf("done initializing plugins")
 }
 
+func ShutdownPlugins() {
+	Events().EmitWithCallback(ShutdownEventSelector, ShutdownEvent{"apid is going to shutdown"}, ShutdownHandler)
+}
+
+
+func ShutdownHandler(event Event) {
```

Comment:
I'd recommend making this an anonymous function inside the ShutdownPlugins() function instead of exposing it.